### PR TITLE
Introduce tooling support tests

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Implements #1429 by creating `platform-tooling-support-tests` project and integrating it as an opt-in build step into the main build process. To activate the tests, pass `platform.tooling.support.tests.enabled=true` as a system property.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests)
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: platform-tooling-support-tests/platform-tooling-support-tests.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+dependencies {
+	implementation('commons-io:commons-io:2.6') {
+		because 'downloading tool archives from URL, moving and deleting directory trees'
+	}
+
+	testImplementation(project(':junit-jupiter-api'))
+
+	testRuntimeOnly(project(':junit-platform-console-standalone')) {
+		because 'needed by Ant projects using native junitlauncher task'
```
Comment (sormuras) [inline platform-tooling-support-tests/platform-tooling-support-tests.gradle]: Should not be on the test runtime classpath!

How do I declare that `platform-tooling-support-tests` depends on `junit-platform-console-standalone` library to be built?

File: documentation/src/docs/asciidoc/release-notes/release-notes-5.3.0-M1.adoc [hunk 2]
```diff
@@ -29,7 +29,9 @@ on GitHub.
 
 * New overloaded variant of `isAnnotated()` in `AnnotationSupport` that accepts
   `Optional<? extends AnnotatedElement>` instead of `AnnotatedElement`.
-
+* New integration test sub-project `platform-tooling-support-tests` added. Set system
+  property `platform.tooling.support.tests.enabled` to `true` to activate it in the
+  main build process.
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/release-notes/release-notes-5.3.0-M1.adoc]: We usually don't include non user facing changes in the release notes.

File: platform-tooling-support-tests/projects/gradle-missing-engine/build.gradle [hunk 3]
```diff
@@ -10,13 +10,10 @@ file('file.encoding.txt').write(System.getProperty('file.encoding'))
...
 repositories {
 	mavenLocal()
-	mavenCentral()
 }
 
 dependencies {
-	testImplementation('org.junit.jupiter:junit-jupiter-api:5.2.0') {
-		because 'needed to compile Jupiter tests // TODO use current junit-jupiter-api'
-	}
+	testImplementation('org.junit.jupiter:junit-jupiter-api:5.3.0-SNAPSHOT')
```
Comment (marcphilipp) [inline platform-tooling-support-tests/projects/gradle-missing-engine/build.gradle]: I think we should pass the current version from `:platform-tooling-support-tests:test` to these dependent builds, e.g. by using a system property.

Comment (sormuras) [reply platform-tooling-support-tests/projects/gradle-missing-engine/build.gradle]: Aye. Already almost implemented ... locally.

File: documentation/src/docs/asciidoc/release-notes/release-notes-5.3.0-M1.adoc (hunk 2 above)
Comment (sbrannen) [reply documentation/src/docs/asciidoc/release-notes/release-notes-5.3.0-M1.adoc]: +1

Comment (sormuras) [reply documentation/src/docs/asciidoc/release-notes/release-notes-5.3.0-M1.adoc]: I'll remove 'em.

**Pull request merged**